Travis - Maven 3.6.2 causes build failures due to repository resolution issues

Travis builds are all failing because we can't find the hypertable or async mongo client libraries, both of which are in niche maven repos. Looks like the Travis folks have some clever maven mirroring set up again that is thus falling over.

For completeness here is the error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.5.5:single (default) on project ycsb: Failed to create assembly: Unable to resolve dependencies for assembly 'package': Failed to resolve dependencies for assembly: Missing:
[ERROR] ----------
[ERROR] 1) org.hypertable:hypertable:jar:0.9.5.6
[ERROR] 
[ERROR]   Try downloading the file manually from the project website.
[ERROR] 
[ERROR]   Then, install it using the command: 
[ERROR]       mvn install:install-file -DgroupId=org.hypertable -DartifactId=hypertable -Dversion=0.9.5.6 -Dpackaging=jar -Dfile=/path/to/file
[ERROR] 
[ERROR]   Alternatively, if you host your own repository you can deploy the file there: 
[ERROR]       mvn deploy:deploy-file -DgroupId=org.hypertable -DartifactId=hypertable -Dversion=0.9.5.6 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR] 
[ERROR]   Path to dependency: 
[ERROR]   	1) site.ycsb:ycsb:pom:0.18.0-SNAPSHOT
[ERROR]   	2) site.ycsb:hypertable-binding:jar:0.18.0-SNAPSHOT
[ERROR]   	3) org.hypertable:hypertable:jar:0.9.5.6
[ERROR] 
[ERROR] 2) com.allanbank:mongodb-async-driver:jar:2.0.1
[ERROR] 
[ERROR]   Try downloading the file manually from the project website.
[ERROR] 
[ERROR]   Then, install it using the command: 
[ERROR]       mvn install:install-file -DgroupId=com.allanbank -DartifactId=mongodb-async-driver -Dversion=2.0.1 -Dpackaging=jar -Dfile=/path/to/file
[ERROR] 
[ERROR]   Alternatively, if you host your own repository you can deploy the file there: 
[ERROR]       mvn deploy:deploy-file -DgroupId=com.allanbank -DartifactId=mongodb-async-driver -Dversion=2.0.1 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR] 
[ERROR]   Path to dependency: 
[ERROR]   	1) site.ycsb:ycsb:pom:0.18.0-SNAPSHOT
[ERROR]   	2) site.ycsb:mongodb-binding:jar:0.18.0-SNAPSHOT
[ERROR]   	3) com.allanbank:mongodb-async-driver:jar:2.0.1
[ERROR] 
[ERROR] ----------
[ERROR] 2 required artifacts are missing.
[ERROR] 
[ERROR] for artifact: 
[ERROR]   site.ycsb:ycsb:pom:0.18.0-SNAPSHOT
[ERROR] 
[ERROR] from the specified remote repositories:
[ERROR]   google-maven-central (https://maven-central.storage-download.googleapis.com/maven2/, releases=true, snapshots=false),
[ERROR]   sonatype (https://oss.sonatype.org/content/repositories/releases/, releases=true, snapshots=false),
[ERROR]   sonatype-snapshots (https://oss.sonatype.org/content/repositories/snapshots/, releases=false, snapshots=true),
[ERROR]   sonatype-apache (https://repository.apache.org/content/repositories/releases/, releases=true, snapshots=false),
[ERROR]   apache-snapshots (https://repository.apache.org/snapshots/, releases=false, snapshots=true)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :ycsb
The command "mvn install -q -DskipTests=true" failed and exited with 1 during 
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -30,15 +30,16 @@ addons:
   hostname: myshorthost
   postgresql: "9.5"
 
-install: mvn install -q -DskipTests=true
+install:
+  - mvn -N io.takari:maven:0.7.7:wrapper -Dmaven=3.6.3
+  - ./mvnw install -q -DskipTests=true
 
-script: mvn test -q
+script: ./mvnw test -q
 
 before_script:
   - psql -c 'CREATE database test;' -U postgres
   - psql -c 'CREATE TABLE usertable (YCSB_KEY VARCHAR(255) PRIMARY KEY not NULL, YCSB_VALUE JSONB not NULL);' -U postgres -d test
   - psql -c 'GRANT ALL PRIVILEGES ON DATABASE test to postgres;' -U postgres
-
 
 # Services to start for tests.
 services:
@@ -47,7 +48,6 @@ services:
   - postgresql
 # temporarily disable riak. failing, docs offline.
 #  - riak
-
 
 # Can't use container based infra because of hosts/hostname
 sudo: true

--- a/pom.xml
+++ b/pom.xml
@@ -237,7 +237,11 @@ LICENSE file.
             <configuration>
               <rules>
                 <requireMavenVersion>
-                  <version>3.1.0</version>
+                  <!--
+                  Maven 3.6.2 has issues
+                  https://github.com/brianfrankcooper/YCSB/issues/1390
+                  -->
+                  <version>[3.1.0,3.6.2),(3.6.2,)</version>
                 </requireMavenVersion>
               </rules>
             </configuration>

